### PR TITLE
Remove support for GH_ORIENTATION in LFRic API (for #960)

### PR DESCRIPTION
**Pull request by TeranIvy**
The changes in this PR are not large but they spread amongst several modules and tests. I deleted the test kernels and examples for `GH_ORIENTATION` and modified the Python tests accordingly. Some of the tests were obviously linked to the orientation support so they were removed. Others were just using orientation kernels and test examples for convenience so those were modifed, hopefully in a correct way to preserve their original purpose.

Tests with compilation pass. Since some of the modifed tests cover annexed dofs and asynchronous halo exchanges, it would be greta if @rupertford could review this PR.

### CODE REVIEW

**Comment by TeranIvy** [comment]
Taking the branch back for a moment to update it.

**Review by rupertford — CHANGES_REQUESTED** [review]
Nicely done.
Just a couple of very minor changes needed and a question. Please see inline comments.

File: doc/user_guide/dynamo0p3.rst [hunk 1]
```diff
@@ -1505,72 +1505,63 @@ conventions, are:
...
-         |               |    3      | W0, W1, W2trace, W2Htrace,         |
-         |               |           | W2Vtrace, W3, Wtheta, Wchi         |
-         +---------------+-----------+------------------------------------+
-
-      2) If it is an orientation array, include the associated argument.
-         The argument is an integer array of type ``i_def`` with intent ``in``.
-         There is one dimension of size the local degrees of freedom for the
-         function space. The name of the array is
-         ``"orientation_"<field_function_space>``.
+   3) For each operation on the function space (``basis``, ``diff_basis``),
```
[the remark this answers is not in the document]
Comment (TeranIvy) [reply doc/user_guide/dynamo0p3.rst:1508]: Oops! Done.

File: src/psyclone/domain/lfric/function_space.py [hunk 2]
```diff
@@ -114,7 +114,7 @@ class FunctionSpace(object):
     VALID_EVALUATOR_NAMES = ["gh_basis", "gh_diff_basis"]
 
     # Meta functions
-    VALID_METAFUNC_NAMES = VALID_EVALUATOR_NAMES + ["gh_orientation"]
+    VALID_METAFUNC_NAMES = VALID_EVALUATOR_NAMES
```
[the remark this answers is not in the document]
Comment (TeranIvy) [reply src/psyclone/domain/lfric/function_space.py:117]: Yes, evaluators are a subset of metafuncs. Metafuncs could include other contents in future so I think is better to have separate names.

File: src/psyclone/dynamo0p3.py [hunk 3]
```diff
@@ -7612,7 +7534,7 @@ def gen_stub(self):
         for entities in [DynCellIterators, DynDofmaps, DynFunctionSpaces,
                          DynCMAOperators, DynScalarArgs, DynFields,
                          DynLMAOperators, DynStencils, DynBasisFunctions,
-                         DynOrientations, DynBoundaryConditions,
+                         DynBoundaryConditions,
```
[the remark this answers is not in the document]
Comment (TeranIvy) [reply src/psyclone/dynamo0p3.py:7537]: As a nitpicky person I completely agree 😃 Removed.

File: src/psyclone/tests/dependency_test.py [hunk 4]
```diff
@@ -668,31 +667,6 @@ def test_lfric_stub_args2():
     assert "weights_xyz_qr_face: READ" in var_info
 
 
-def test_lfric_stub_args3():
```
[the remark this answers is not in the document]
Comment (TeranIvy) [reply src/psyclone/tests/dependency_test.py:671]: Yes, you're right. I reinstated it and modified it to work with `testkern_any_discontinuous_space_op_1_mod.f90` that also has a "read" and a "write" operator.

File: src/psyclone/dynamo0p3.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (TeranIvy) [reply src/psyclone/dynamo0p3.py:7537]: There were some `pylint` errors in `dynamo0p3.py` from a recent change on master (wrong indentation) so I fixed that, too.

**Comment by TeranIvy** [comment]
Addressed review comments and merged the latest master. Ready for another review from @rupertford.

File: src/psyclone/tests/dependency_test.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (rupertford) [reply src/psyclone/tests/dependency_test.py]: Great, thanks.

**Review by rupertford — APPROVED** [review]
All issues have been addressed.
documentation builds and changes look good
all changes conform to pycodestyle and pylint
Approving.

**Pull request merged**